lib/compatibility/BUILD

```diff
@@ -0,0 +1,32 @@
+load(":defs.bzl", "MAX_NUM_ALL_OF_SETTINGS")
+load("//:bzl_library.bzl", "bzl_library")
+
+licenses(["notice"])
+
+package(default_visibility = ["//visibility:public"])
+
+bzl_library(
+    name = "defs",
+    srcs = ["defs.bzl"],
+)
+
+filegroup(
+    name = "build_file",
+    testonly = True,
+    srcs = ["BUILD"],
+)
+
+constraint_value(
+    name = "none_of",
+    constraint_setting = "@platforms//:incompatible_setting",
+)
+
+constraint_value(
+    name = "any_of",
+    constraint_setting = "@platforms//:incompatible_setting",
+)
```

Comment:
The name `any_of` is confusing. Reading the code for `_any_of`, I had no idea it works until I looked up how `any_of` is actually implemented.

I can see now, you named constraints by the location where they are used, but that's confusing. 

Is there a reason you're defining new `constraint_values` and not using directly `@platforms//:incompatible`, which is a constraint value already?

If not, I'd suggest you just remove all `constraint_value`-s and use the predefined one in the functions.

Reply:
This goes back to the earlier discussion: https://github.com/bazelbuild/bazel-skylib/pull/381#discussion_r954544755

The short version is that bazel is not happy about duplicated labels in a label list.

Reply:
To elaborate a little bit: In today's bazel, you cannot freely compose everything with `@platforms//:incompatible` because you risk getting the label into the final label list more than once. Bazel will then error out.

Reply:
Ack. Sorry, I forgot about that. Can you then rename to something like `incompatible_in_any_of`? And please "sprinkle" some comments about this problem :)